internal/graph/graph.go

```diff
@@ -293,15 +377,87 @@ func (g *ConnectedObjectGraph) findIngressesWithTargetRewrite(
 		var res []*RelationshipIngress
 		for _, child := range t.Union.GetChild() {
 			// we recurse through each child rewrite
-			childResults, err := g.findIngressesWithTargetRewrite(target, source, child, visited)
+			childResults, err := g.findIngressesWithTargetRewrite(target, source, child, visited, findIngressOption)
 			if err != nil {
 				return nil, err
 			}
 			res = append(res, childResults...)
 		}
 		return res, nil
-	case *openfgapb.Userset_Intersection, *openfgapb.Userset_Difference:
-		return nil, ErrNotImplemented
+	case *openfgapb.Userset_Intersection:
+
+		if findIngressOption == resolveAnyIngress {
+			child := t.Intersection.GetChild()[0]
+
+			childresults, err := g.findIngressesWithTargetRewrite(target, source, child, visited, findIngressOption)
+			if err != nil {
+				return nil, err
+			}
+
+			for _, childresult := range childresults {
+				childresult.Condition = RequiresFurtherEvalCondition
+			}
+
+			return childresults, nil
+		}
+
+		var ingresses []*RelationshipIngress
+		for _, child := range t.Intersection.GetChild() {
+
+			res, err := g.findIngressesWithTargetRewrite(target, source, child, visited, findIngressOption)
+			if err != nil {
+				return nil, err
+			}
+
+			ingresses = append(ingresses, res...)
+		}
+
+		if len(ingresses) > 0 {
+			ingresses[0].Condition = RequiresFurtherEvalCondition
```

Comment:
Can you expand a bit about this? Why are we setting RequiresFurtherEvalCondition only for the first ingress?

Reply:
As we discussed offline, the handling of ingress resolution for all ingresses (e.g. when we use `resolveAllIngresses`) is  non-trivial for cases involving intersection and exclusion. We don't today have a great canonical representation for these kinds of ingress structures.

The right approach to handling this in the future would be to represent ingresses using an AST-like structure that represents expression under intersection or exclusion as an AST sub-tree whose structural representation allows for an ingress that has a conditional status of `RequiresFurtherEvaluationCondition` to be defined on the parent AST node of the expression. Since we do not yet have this representation, and we have no use for resolving all relationship ingresses (given the current state of the ConnectedObjects algorithm), then this work should be done as a follow up when this functionality is needed. I will open an issue for this describing it in more detail, but for now we'll mark this as a non-blocker and proceed.